### ISSUE TITLE
Synchronise keys with parent when checking entitlements.

### ISSUE THREAD

**Issue by timbru** [NLnetLabs/krill#499]
Krill CAs check entitlements with their parents for differences in resources and the 'not after time' on certificates.

But they assume that their *current* key is also the known *current* key from the parent's point of view.

This can lead to issues if a key roll goes wrong somehow, or if there is restore from backup. There should not ever be a discrepancy, but theoretically there could be.

This issue then, is to make sure the child CAs verifies that the known keys by the parent match what it believes it has. And if there is any difference, then take action to remedy..

Essentially revoke all keys at the parent unknown to the child (this is done already) and make sure that the child gets a new active key. This needs some thought but the easiest is probably to ask for revocation of all keys that the child has, and start over with a new key and request a certificate for it.

**Comment by timbru** [comment]
Reading RFC 6492 once more.. (with feeling..) it seems that even though all 'current' certificates issued to a child "are to be included" in a Resource Class List Response - there is no normative MUST in the text that will hold parent implementation to this:  https://datatracker.ietf.org/doc/html/rfc6492#section-3.3.2

In short we cannot trust that we will **always** get all current certificates, i.e. on for each key. It is very likely that we do, but not certain.

Therefore we should be a bit more careful here. It is okay to ask for the revocation of surplus keys - i.e. we see certificates for keys that we do not know. But the absence of a certificate for each of the keys in the CAs resource class does not necessarily mean that we can't have a certificate for those.

So long story short, we should just request updated certificates when resource or validity entitlement change, like we do now. What would be very good to do and in scope of this issue though, is to make Krill CAs re-initialise the key state if it gets a "1204  request - already used key in request" error in response to a Certificate Issuance Request: https://datatracker.ietf.org/doc/html/rfc6492#section-3.4.1
